Simple Incremental/Assets/Scripts/EnemyHook.cs

```diff
@@ -14,12 +14,15 @@ public void ChangeTemplate(EnemyTemplate enemyTemplate)
             SpriteRenderer spriteRenderer = GetComponent<SpriteRenderer>();
             CharacterHealth characterHealth = GetComponent<CharacterHealth>();
             CharacterLoot characterLoot = GetComponent<CharacterLoot>();
+            Movement movement = GetComponent<Movement>();
             if(spriteRenderer != null)
                 spriteRenderer.sprite = enemyTemplate.sprite;
             if(characterHealth != null)
                 characterHealth.maxHealth = enemyTemplate.health;
             if(characterLoot != null)
                 characterLoot.coins = enemyTemplate.coins;
+            if (movement != null)
+                movement.moveSpeed = enemyTemplate.moveSpeed;
```

Comment:
Since we don't appear to be making use of any of these variables would it makes sense to simplify this section to something like:
```
GetComponent<SpriteRenderer>()?.sprite = enemyTemplate.sprite;
GetComponent<CharacterHealth>()?.maxHealth = enemyTemplate.health;
GetComponent<CharacterLoot>()?.coins = enemyTemplate.coins;
GetComponent<Movement>()?.moveSpeed = enemyTemplate.moveSpeed;
```

Reply:
Absolutely. You are my favorite for that. I was getting annoyed with how verbose it was getting.

Reply:
SO! Partial solution, unfortunately the GetComponent can't be nullchecked on the spot. It has to be stored to a variable.